### ISSUE TITLE
Node for parenthesized expressions

### ISSUE THREAD

**Issue by HiPhish** [DerekStride/tree-sitter-sql#274]
Hello,

maintainer of [HiPhish/rainbow-delimiters.nvim](https://github.com/hiphish/rainbow-delimiters.nvim) here; I have a problem matching and properly highlighting parenthesized expressions. The problem is that all the parentheses of an expression are on the same level as their siblings. Consider the following query:

```sql
SELECT ((((1))));
```

It's a weird query, but at least in SQLite it is valid. The tree looks like this:

```
  (select
    (keyword_select)
    (select_expression
      (term
        value: "("
        value: "("
        value: "("
        value: "("
        value: (literal)
        value: ")"
        value: ")"
        value: ")"
        value: ")")))
```

All the parentheses are on the same level, it is impossible to build up a tree of "containers" to determine the proper nesting. I propose a sort of `parenthesized_expression` node (or whatever name you consider more appropriate). With that node the `term` would have only one `value`:

```
  (select
    (keyword_select)
    (select_expression
      (term
    	value: (parenthesized_expression
    	  "("
    	  (parenthesized_expression
    	    "("
    	    (parenthesized_expression
    	      "("
    	      (parenthesized_expression
    	        "("
                (literal)
    	        ")")
    	      ")")
    	    ")")
    	  ")"))))
```

Let's take another example:

```sql
SELECT 1 + (2);
```

The original tree:

```
  (select
    (keyword_select)
    (select_expression
      (term
        value: (binary_expression
          left: (literal)
          operator: "+"
          right: "("
          right: (literal)
          right: ")"))))
```

How many `right` nodes can a binary expression reasonably have? Here is what it would look like with `parenthesized_expression`:

```
(select
  (keyword_select)
  (select_expression
    (term
      value: (binary_expression
        left: (literal)
        operator: "+"
        right: (parenthesized_expression
          "("
          (literal)
          ")")))))
```

**Comment by matthias-Q** [comment]
I agree. I think our AST is sometimes way to "flat". I will take a look later today.

**Comment by matthias-Q** [comment]
Closed by #275 

Please tell me if that change is sufficient.

**Comment by HiPhish** [comment]
Looks great, thank you!

![Screenshot_20240831_002856](https://github.com/user-attachments/assets/3f409844-7245-493a-acb0-c5713de062c4)
